ledger/participant-integration-api/src/main/scala/platform/store/dao/events/QueryNonPruned.scala

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.dao.events
+
+import java.sql.Connection
+
+import anorm.SQL
+import com.daml.ledger.participant.state.v1.Offset
+import com.daml.platform.server.api.validation.ErrorFactories
+import com.daml.platform.store.Conversions.offset
+
+object QueryNonPruned {
```

Comment:
In the four streaming queries we want to produce an error rather than filtering the results, as that is the mechanism by which we make users aware of a potentially broken app (that is either unaware of pruning and the resulting inability to read the ledger from `begin` or has fallen behind the pruning schedule).

However in the two point-id queries, we do filter, and those cases we could handle by doing the join in the database.

Reply:
I see. thanks.